fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/codec/PrefetchableBufferedChecksumIndexInput.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * PrefetchableBufferedChecksumIndexInput.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.lucene.codec;
+
+import com.apple.foundationdb.record.lucene.directory.FDBIndexInput;
+import org.apache.lucene.store.BufferedChecksumIndexInput;
+import org.apache.lucene.store.IndexInput;
+
+import java.io.IOException;
+
+/**
+ * An Index Input that attempts to keep a 10 block buffer in front or at the read at a minimum.  The goal for this
+ * class is to provide some parallelism for fetches from FDB when we are attempting to merge segments.  The signaling
+ * of when to prefetch occurs in the FDBIndexOutput#copyBytes method.
+ *
+ */
+public class PrefetchableBufferedChecksumIndexInput extends BufferedChecksumIndexInput {
+    private final IndexInput indexInput;
+    private static final int PREFETCH_BLOCKS = 10; // Attempt to keep a large enough buffer
+    /**
+     * GreatedFetchBlock from the underlying FDBIndexInput, -1 if not fetched.
+     *
+     */
+    private int greatestFetchedBlock;
+    /**
+     * Maximum fetched block based on the numberOfBytesToPrefetch.
+     *
+     */
+    private int maximumFetchedBlock;
+    /**
+     * Checks whether it is an instance of FDBIndexInput.
+     *
+     */
+    private final boolean supportsPrefetch;
+
+    public PrefetchableBufferedChecksumIndexInput(final IndexInput indexInput) {
+        super(indexInput);
+        this.indexInput = indexInput;
+        this.supportsPrefetch =  indexInput instanceof FDBIndexInput;
+    }
+
+    public void setNumberOfBytesToPrefetch(long numberOfBytesToPrefetch) {
```

Comment:
I think it's a bit confusing to have this method called `setNumberOfBytesToPrefetch` (which makes it sound like all it does is set a member variable). I think something like `prefetch` or `prefetchBytes` might be clearer